cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+
+- Users should decide whether to actually create the cluster, or print a manifest only.
+
+- A **cluster name must be provided by the user**. We won't generate a cluster name for the user.
+  - At Giant Swarm we decided around November 2023 that we want users to set workload cluster names and avoid auto-generated names ([roadmap#2999](https://github.com/giantswarm/roadmap/issues/2999)). In theory, we could still offer an option (a flag) to auto-generate the name. However, this would require knowledge of the cluster-PROVIDER app schema in the command logic. For example, as of cluster-aws v0.73.0, the cluster name is set in the property .global.metadata.name. Although we are working towards alignment across providers, this may be different in other versions or other provider apps.
+
+## Basic syntax
+
+The command should be named `create cluster`, hence the entire command, without any flags, would be:
+
+    kubectl gs create cluster
+
+The most important flags would be:
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--app-version`: the cluster-PROVIDER app version to use. If not specified, the latest version is used.
+- `--dry-run`: only validate the configuration and print the resulting manifest.
+- `--output`: file path to write the result to. If not specified, resources are written to the management cluster. The option `STDOUT` can be used to print the result to the console.
+- `--format`: This flag can be used to specify the format of the output.
+  - Default is `manifest` for a Kubernetes manifest that includes an App resources and one or several ConfigMap resources.
+  - `config` for merged configuration YAML.
+  - `command` for a self-contained representation of the command just executed, as a one-liner.
+- `--set`: Override a configuration value. This flag can be used multiple times. The format is `path.to.property=value`.
+  - Note: `--set` can only be used with scalar values (string, number, boolean, integer, etc.). It cannot be used to override an array or object.
+
+TODO:
+
+- How should the resource **namespace** be specified? It could be derived from the organization name included in the cluster config, however that would require logical understanding of the config, which may differ over time and between providers.
+
+### Examples
+
+Ex. 1: Simple cluster creation based on a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml
+
+Ex. 2: Simple cluster creation based on standard input
+
+    kubectl gs create cluster --provider capa <<EOF
+    global:
+      metadata:
+        servicePriority: lowest
+        name: test01
+        organization: testorg
+        description: Just a test cluster
```

Comment:
In an ideal case, the CLI wouldn't require passing any config unless the user wants to override defaults.

For example, it would be very friendly if this were a correct and complete way to start a cluster:
```
kubectl gs create cluster --provider capa --name test01
```

Based on the discussion in https://github.com/giantswarm/roadmap/issues/2999, `name` will be required and can be supplied via the CLI, but IIUC the others listed here could be defaulted (priority, org) or left empty (description). Is that true?

Reply:
I agress on the approach of having a valid config by providing minimal config, however, there will be a few infra-specific parameters (really not a lot) for the onprem providers that we can't default.

Reply:
@stone-z What you describe is really the intended behaviour. We aim to ask only the required details from the users. However, with `kubectl gs create cluster --provider capa --name test01`, the one thing I'm missing is the organization. How can we set a default for organization (and thus namespace)?

Reply:
I'd not add a default org for now and it would also be a bit weird when a user is a member of several orgs but not all. Only in the single org member case we can clearly set a default.

Reply:
Honestly, thinking of this as a kubectl command, kubectl has a default/selected namespace, I kinda feel we should stay true to that if at all.

Reply:
See also my comment https://github.com/giantswarm/rfc/pull/98#discussion_r1627104913

Reply:
After some talking in todays KaaS sync I went with the `--namespace` flag. I dislike that it's one more flag the user will have to set, as "default" will result in non-functioning clusters, but all other solutions seem too involved for now.